examples/pytorch/infograph/unsupervised.py

```diff
@@ -0,0 +1,146 @@
+import torch as th
+
+import dgl
+from dgl.data import GINDataset
+from dgl.dataloading import GraphDataLoader
+
+from model import InfoGraph
+from evaluate_embedding import evaluate_embedding
+
+import argparse
+
+
+def argument():
+    parser = argparse.ArgumentParser(description='InfoGraph')
+    # data source params
+    parser.add_argument('--dataname', type=str, default='MUTAG', help='Name of dataset.')
+
+    # training params
+    parser.add_argument('--gpu', type=int, default=-1, help='GPU index, default:-1, using CPU.')
+    parser.add_argument('--epochs', type=int, default=20, help='Training epochs.')
+    parser.add_argument('--batch_size', type=int, default=128, help='Training batch size.')
+    parser.add_argument('--lr', type=float, default=0.01, help='Learning rate.')
+
+    # model params
+    parser.add_argument('--n_layers', type=int, default=3, help='Number of graph convolution layers before each pooling')
+    parser.add_argument('--hid_dim', type=int, default=32, help='Hidden layer dimensionalities')
+
+    args = parser.parse_args()
+
+    # check cuda
+    if args.gpu != -1 and th.cuda.is_available():
+        args.device = 'cuda:{}'.format(args.gpu)
+    else:
+        args.device = 'cpu'
+
+    return args
+
+    
+def collate(samples):
+    ''' collate function for building graph dataloader'''
+    graphs, labels = map(list, zip(*samples))
+
+    # generate batched graphs and labels
+    batched_graph = dgl.batch(graphs)
+    batched_labels = th.tensor(labels)
+
+    n_nodes = batched_graph.num_nodes()
+
+    # generate graph_id for each node within the batch
+    graph_id = th.zeros(n_nodes).long()
+    N = 0
+    id = 0
+    for graph in graphs:
+        N_next = N + graph.num_nodes()
+        graph_id[N:N_next] = id
+        N = N_next
+        id += 1
```

Comment:
You can use [broadcast_nodes](https://docs.dgl.ai/generated/dgl.broadcast_nodes.html#dgl-broadcast-nodes). I'm not sure whether this is very efficient though.